Dockerfile

```diff
@@ -0,0 +1,42 @@
+FROM ubuntu
+
+ENV DEBIAN_FRONTEND=noninteractive
+RUN apt-get update && \
+    apt-get install -y \
+        libffi6 pkg-config libffi-dev pkg-config build-essential \
+        gcc git wget make libncurses-dev flex bison gperf autoconf \
+        python3 python3-dev python3-pip python python-dev python-pip \
+        make unrar-free autoconf automake libtool gcc g++ gperf \
+        flex bison texinfo gawk ncurses-dev libexpat-dev \
+        sed git unzip bash help2man wget bzip2 \
+        libtool-bin gcc-arm-none-eabi qemu-system-arm
+
+RUN useradd -m docker
+USER docker
+ENV HOME /home/docker
+WORKDIR $HOME
+
+# setup esp toolchain
+RUN mkdir -p $HOME/esp
+WORKDIR $HOME/esp
+RUN wget -qO- https://dl.espressif.com/dl/xtensa-esp32-elf-linux64-1.22.0-80-g6c4433a-5.2.0.tar.gz | tar -xzf -
+ENV PATH "$HOME/esp/xtensa-esp32-elf/bin:$PATH"
```

Comment:
The version of this needs to be linked to the version required by esp port

Reply:
How do I find this?